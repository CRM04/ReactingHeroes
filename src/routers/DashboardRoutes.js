import React from 'react'
import { Navbar } from '../components/ui/Navar'
import { Route, Switch, Redirect } from 'react-router-dom';
import MarvelScreen from '../components/marvel/MarvelScreen';
import HeroesScreen from '../components/heroes/HeroesScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mx-auto mt-5">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen}></Route>
                    <Route exact path="/dc" component={DcScreen}></Route>
                    <Route exact path="/heroe/:heroeId" component={HeroesScreen}></Route>
                    <Route exact path="/search" component={SearchScreen}></Route>
                    <Redirect to="/marvel"></Redirect>
                </Switch>
            </div>
        </>
    )
}
