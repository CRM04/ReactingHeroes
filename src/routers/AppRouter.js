import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import loginScreen from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

const AppRouter = () => {
    const { user } = useContext(AuthContext);

    return (
        <Router>
            <div>
                <Switch>
                    {/* <Route path="/login" exact component={loginScreen}></Route> */}
                    {/* <Route path="/" component={DashboardRoutes}></Route> */}
                    <PublicRoute  path='/login' component={loginScreen} isAuth={user.logged}></PublicRoute>
                    <PrivateRoute path='/' component={DashboardRoutes} isAuth={user.logged}></PrivateRoute>
                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter
