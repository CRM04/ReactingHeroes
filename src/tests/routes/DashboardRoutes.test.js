import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import { AuthContext } from "../../auth/AuthContext";

describe('<DashboardRoutes/> test suit', () => {

    test('should show the right components route', () => {
        const context = {
            user: {
                logged: true,
                name: 'Juan'
            },
            dispatch: jest.fn
        }
        const wrapper = mount(
            <AuthContext.Provider value={context}>
                <MemoryRouter>
                    <DashboardRoutes></DashboardRoutes>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("#userNameNav").text()).toBe('Juan')
    });


});