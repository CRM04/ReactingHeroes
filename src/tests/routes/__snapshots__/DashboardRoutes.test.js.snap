// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardRoutes/> test suit should show the right components route 1`] = `
Array [
  <div
    className=""
  >
    <nav
      className="bg-navBar shadow-2xl"
    >
      <div
        className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 "
      >
        <div
          className="flex sm:justify-between lg:justify-start"
        >
          <div
            className="flex items-center"
          >
            <div
              className=""
            >
              <img
                alt="Workflow"
                className="h-14 w-auto"
                src="../assets/heroes/logo.png"
              />
            </div>
            <div
              className="hidden md:flex justify-between"
            >
              <div
                className="ml-10 flex items-baseline space-x-4"
              >
                <a
                  aria-current="page"
                  className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/marvel"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Marvel
                </a>
                <a
                  aria-current={null}
                  className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/dc"
                  onClick={[Function]}
                >
                  DC
                </a>
                <a
                  aria-current={null}
                  className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/search"
                  onClick={[Function]}
                >
                  Search
                </a>
              </div>
              <div
                className=" flex items-baseline space-x-4"
              >
                <p
                  className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                  id="userNameNav"
                >
                  Juan
                </p>
                <button
                  className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                  id="logout"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
          </div>
          <div
            className="-mr-2 flex md:hidden"
          >
            <button
              aria-controls="mobile-menu"
              aria-expanded="false"
              className="inline-flex items-center justify-center pt-1 pb-1 pr-2 pl-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
              onClick={[Function]}
              type="button"
            >
              <span
                className="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                className="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="container mx-auto mt-5"
  >
    <div>
      <div
        className="flex justify-around flex-wrap content-start items-start animate__animated animate__fadeIn"
      >
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Spider Man"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-spider.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Spider Man
                </h4>
                <div
                  className="text-gray-800"
                >
                  Peter Parker
                </div>
                <small
                  className="text-gray-500"
                >
                  Amazing Fantasy #15
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-spider"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Captain America"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-captain.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Captain America
                </h4>
                <div
                  className="text-gray-800"
                >
                  Steve Rogers
                </div>
                <small
                  className="text-gray-500"
                >
                  Captain America Comics #1
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-captain"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Iron Man"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-iron.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Iron Man
                </h4>
                <div
                  className="text-gray-800"
                >
                  Tony Stark
                </div>
                <small
                  className="text-gray-500"
                >
                  Tales of Suspense #39
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-iron"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Thor"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-thor.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Thor
                </h4>
                <div
                  className="text-gray-800"
                >
                  Thor Odinson
                </div>
                <small
                  className="text-gray-500"
                >
                  Journey into Myster #83
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-thor"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Hulk"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-hulk.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Hulk
                </h4>
                <div
                  className="text-gray-800"
                >
                  Bruce Banner
                </div>
                <small
                  className="text-gray-500"
                >
                  The Incredible Hulk #1
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-hulk"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Wolverine"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-wolverine.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Wolverine
                </h4>
                <div
                  className="text-gray-800"
                >
                  James Howlett
                </div>
                <small
                  className="text-gray-500"
                >
                  The Incredible Hulk #180
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-wolverine"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Daredevil"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-daredevil.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Daredevil
                </h4>
                <div
                  className="text-gray-800"
                >
                  Matthew Michael Murdock
                </div>
                <small
                  className="text-gray-500"
                >
                  Daredevil #1
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-daredevil"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Hawkeye"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-hawkeye.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Hawkeye
                </h4>
                <div
                  className="text-gray-800"
                >
                  Clinton Francis Barton
                </div>
                <small
                  className="text-gray-500"
                >
                  Tales of Suspense #57
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Cyclops"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-cyclops.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Cyclops
                </h4>
                <div
                  className="text-gray-800"
                >
                  Scott Summers
                </div>
                <small
                  className="text-gray-500"
                >
                  X-Men #1
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-cyclops"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
        <div
          className="shadow border-gray-200 m-1 h-auto rounded lg:w-1/5 sm:w1 md:w-1/3 hover: transform-gpu hover:-translate-y-2 ease-linear duration-75 mt-10"
        >
          <div
            className="lg:flex sm:block "
          >
            <div
              className="bg-blue-800  w-100 h-100 rounded-tl rounded-tr"
            />
            <img
              alt="Silver Surfer"
              className="lg:w-1/2 sm:w1 rounded-tl rounded-bl hover:filter hover:grayscale "
              src="assets/heroes/marvel-silver.jpg"
            />
            <div
              className="flex flex-col flex-auto justify-between"
            >
              <div
                className="p-2"
              >
                <h4
                  className="font-semibold text-lg uppercase "
                >
                  Silver Surfer
                </h4>
                <div
                  className="text-gray-800"
                >
                  Norrin Radd
                </div>
                <small
                  className="text-gray-500"
                >
                  The Fantastic Four #48
                </small>
              </div>
              <a
                className="p-2 w-auto bg-gray-100 rounded-br  text-red-700 font-semibold hover:bg-red-900 hover:text-white relative bottom-0"
                href="/heroe/marvel-silver"
                onClick={[Function]}
              >
                Ver m치s
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
